KPI-team1-frontend/src/pages/UserDetailsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from "react";
+import { useOutletContext } from "react-router-dom";
+import { Circles } from "../model/circle";
+import { User, UserDetails } from "../model/user";
+
+interface UserUpdateError {
+  message: String;
+}
+
+interface OutletContext {
+  user: User;
+  circles: Circles[];
+  userDetails: UserDetails;
+  setUserDetails: (u: UserDetails) => void;
+}
+
+export default function UserDetailsPage(): JSX.Element {
+  const { circles, user, userDetails, setUserDetails }: OutletContext = useOutletContext();
+
+  const [defaultCircleId, setDefaultCircleId] = useState(userDetails.defaultCircleId);
+
+  const [username, setUsername] = useState(userDetails.username);
+
```

Comment:
to prevent the page from breaking upon refreshing the settings page, it should be checked that `userDetails` is not falsy values  before calling its properties (`defaultCircleId` and `username`) by using optional chaining: 
`const [defaultCircleId, setDefaultCircleId] = useState(userDetails?.defaultCircleId);`
`const [username, setUsername] = useState(userDetails?.username);`

Reply:
Hmmm, if the user is not logged in then he shouldn't see the link to the settings page, but he may in fact just type the url in the browser. Updated.